src/java.compiler/share/classes/javax/tools/SimpleJavaFileObject.java

```diff
@@ -223,4 +223,29 @@ public boolean isNameCompatible(String simpleName, Kind kind) {
     public String toString() {
         return getClass().getName() + "[" + toUri() + "]";
     }
+
+    /**
+     * Creates a {@link JavaFileObject} which will be represents the given source content.
+     *
+     * <p>The provided {@code uri} will be returned from {@link #toUri()}.
+     * The provided {@code content} will be returned from {@link #getCharContent(boolean)}.
+     * The {@link #getKind()} method will return {@link Kind#SOURCE}.
+     *
+     * <p>All other methods will behave as described in the documentation in this class,
+     * if the constructor is called with {@code uri} and {@code Kind.SOURCE}.
+     *
+     * @param uri that should be used for this {@code JavaFileObject}
```

Comment:
I realize this is a static method, but the use of "this" might mean to imply the current instance on which the method is invoked. Perhaps, we should change it to `that should be used by the returned {@code JavaFileObject}` ? Or maybe just replace "this" with "the"?